torch/_subclasses/meta_utils.py

```diff
@@ -235,27 +235,51 @@ def meta_tensor(
             maybe_suppress = shape_env.suppress_guards
 
         def sym_sizes_strides_storage_offset(
-            t, src
+            t, src, policy=policy
         ) -> Tuple[Tuple[int, ...], Tuple[int, ...], int]:
             if shape_env is not None:
-                if isinstance(t, FakeTensor) and t.fake_mode.shape_env is shape_env:
+                # Returns the ShapeEnv from the FakeMode associated with this tensor, or None.
+                def get_fake_mode_shape_env(t):
+                    if isinstance(t, FakeTensor):
+                        return t.fake_mode.shape_env
+
+                    # subclass case: try to pull fake mode from inner tensors
+                    if is_traceable_wrapper_subclass(t):
+                        attrs, _ = t.__tensor_flatten__()
+                        for attr in attrs:
+                            inner = getattr(t, attr)
+                            if isinstance(inner, FakeTensor):
+                                return inner.fake_mode.shape_env
+
+                    return None
```

Comment:
nit: Can this be a util?

Reply:
yep I'll bump it somewhere else - any preference?